src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use DateTimeImmutable;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    /**
+     * @return array[]
+     */
+    public function findActivePromotionsByPid(Pid $pid, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+            ->addSelect(['context', 'promotionOfCoreEntity', 'promotionOfImage'])
+            ->leftJoin('promotion.context', 'context')
+            ->leftJoin('promotion.promotionOfCoreEntity', 'promotionOfCoreEntity')
+            ->leftJoin('promotion.promotionOfImage', 'promotionOfImage')
+            ->andWhere('promotion.isActive = 1')
+            ->andWhere('promotion.startDate <= :now')
+            ->andWhere('promotion.endDate >= :now')
+            ->andWhere('context.pid = :pid')
+            ->addOrderBy('promotion.weighting', 'ASC')
+            ->setFirstResult($offset)
+            ->setMaxResults($limit)
+            ->setParameter('pid', $pid)
+            ->setParameter('now', $datetime);
+
+        return $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
+    }
+
+    /**
+     * @param int[] $ancestryIds
+     * @return array[]
+     */
+    public function findActiveSuperPromotionsByPid(array $ancestryIds, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+           ->addSelect(['context', 'promotionOfCoreEntity', 'promotionOfImage'])
```

Comment:
The same comments as above on 'context, the join and the naming of 'now' also apply here